lib/signex/parameters.ex

```diff
@@ -0,0 +1,48 @@
+defmodule SignEx.Parameters do
+  @moduledoc """
+    A `SignEx.Parameters` struct contains all the information needed to reconstruct the signature of a message.
+  """
+
+  require Logger
+
+  defstruct [:key_id, :algorithm, :headers, :signature]
+
+  def to_string(params) do
+    Logger.warn("Deprechiated: Use `SignEx.Parameters.serialize`")
+    serialize(params)
+  end
+
+  @doc """
+  Serialize parameters as a HTTP signature header value
+  """
+  def serialize(%__MODULE__{
+    key_id: key_id,
+    algorithm: algorithm,
+    headers: headers,
+    signature: signature
+    }) do
+      headers = Enum.join(headers, " ")
+     {:ok, "key_id=\"#{key_id}\",algorithm=\"#{algorithm}\",headers=\"#{headers}\",signature=\"#{signature}\""}
```

Comment:
FYI, this isn't quite right for the `Authorization` header. The string needs to start with the auth scheme (ie.  `Signature `)